### PR TITLE
fix(clerk-js): Remove session cookie when signing out and `onBeforeSetActive`

### PR DESCRIPTION
**Pull request by octoper**
## Description

**Problem:**

Right now when signing out on Next.js with app router, there is a chance that you will get an error as we are clearing the app router cache before removing the session cookie.

**Solution:**

The solution is to move the removal of the session cookie before clearing the app router cache.

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

**Comment by octoper** [comment]
!snapshot

**Comment by clerk-cookie** [comment]
Hey @octoper - the snapshot version command generated the following package versions:
| Package | Version |
| --- | --- |
| @clerk/chrome-extension | 1.0.9-snapshot.v40b4e54 |
| @clerk/clerk-js | 5.2.5-snapshot.v40b4e54 |
| @clerk/clerk-expo | 1.1.1-snapshot.v40b4e54 |
| gatsby-plugin-clerk | 5.0.0-beta.45 |
| @clerk/localizations | 2.4.0-snapshot.v40b4e54 |

Tip: Use the snippet copy button below to quickly install the required packages.
`@clerk/chrome-extension`
```sh
npm i @clerk/chrome-extension@1.0.9-snapshot.v40b4e54 --save-exact
```
`@clerk/clerk-js`
```sh
npm i @clerk/clerk-js@5.2.5-snapshot.v40b4e54 --save-exact
```
`@clerk/clerk-expo`
```sh
npm i @clerk/clerk-expo@1.1.1-snapshot.v40b4e54 --save-exact
```
`gatsby-plugin-clerk`
```sh
npm i gatsby-plugin-clerk@5.0.0-beta.45 --save-exact
```
`@clerk/localizations`
```sh
npm i @clerk/localizations@2.4.0-snapshot.v40b4e54 --save-exact
```

**Comment by octoper** [comment]
!snapshot

**Comment by clerk-cookie** [comment]
Hey @octoper - the snapshot version command generated the following package versions:
| Package | Version |
| --- | --- |
| @clerk/chrome-extension | 1.0.9-snapshot.v3595faf |
| @clerk/clerk-js | 5.2.5-snapshot.v3595faf |
| @clerk/clerk-expo | 1.1.1-snapshot.v3595faf |
| gatsby-plugin-clerk | 5.0.0-beta.45 |
| @clerk/localizations | 2.4.0-snapshot.v3595faf |

Tip: Use the snippet copy button below to quickly install the required packages.
`@clerk/chrome-extension`
```sh
npm i @clerk/chrome-extension@1.0.9-snapshot.v3595faf --save-exact
```
`@clerk/clerk-js`
```sh
npm i @clerk/clerk-js@5.2.5-snapshot.v3595faf --save-exact
```
`@clerk/clerk-expo`
```sh
npm i @clerk/clerk-expo@1.1.1-snapshot.v3595faf --save-exact
```
`gatsby-plugin-clerk`
```sh
npm i gatsby-plugin-clerk@5.0.0-beta.45 --save-exact
```
`@clerk/localizations`
```sh
npm i @clerk/localizations@2.4.0-snapshot.v3595faf --save-exact
```

**Comment by octoper** [comment]
!snapshot

**Comment by clerk-cookie** [comment]
Hey @octoper - the snapshot version command generated the following package versions:
| Package | Version |
| --- | --- |
| @clerk/chrome-extension | 1.0.9-snapshot.vb1a12ac |
| @clerk/clerk-js | 5.2.5-snapshot.vb1a12ac |
| @clerk/clerk-expo | 1.1.1-snapshot.vb1a12ac |
| gatsby-plugin-clerk | 5.0.0-beta.45 |
| @clerk/localizations | 2.4.0-snapshot.vb1a12ac |
| @clerk/nextjs | 5.0.9-snapshot.vb1a12ac |

Tip: Use the snippet copy button below to quickly install the required packages.
`@clerk/chrome-extension`
```sh
npm i @clerk/chrome-extension@1.0.9-snapshot.vb1a12ac --save-exact
```
`@clerk/clerk-js`
```sh
npm i @clerk/clerk-js@5.2.5-snapshot.vb1a12ac --save-exact
```
`@clerk/clerk-expo`
```sh
npm i @clerk/clerk-expo@1.1.1-snapshot.vb1a12ac --save-exact
```
`gatsby-plugin-clerk`
```sh
npm i gatsby-plugin-clerk@5.0.0-beta.45 --save-exact
```
`@clerk/localizations`
```sh
npm i @clerk/localizations@2.4.0-snapshot.vb1a12ac --save-exact
```
`@clerk/nextjs`
```sh
npm i @clerk/nextjs@5.0.9-snapshot.vb1a12ac --save-exact
```

**Comment by octoper** [comment]
!snapshot

**Comment by octoper** [comment]
!snapshot

**Comment by clerk-cookie** [comment]
Hey @octoper - the snapshot version command generated the following package versions:
| Package | Version |
| --- | --- |
| @clerk/backend | 1.1.5-snapshot.v58e76d6 |
| @clerk/chrome-extension | 1.0.10-snapshot.v58e76d6 |
| @clerk/clerk-js | 5.3.1-snapshot.v58e76d6 |
| @clerk/elements | 0.4.2-snapshot.v58e76d6 |
| @clerk/clerk-expo | 1.1.2-snapshot.v58e76d6 |
| @clerk/express | 0.0.8-snapshot.v58e76d6 |
| @clerk/fastify | 1.0.7-snapshot.v58e76d6 |
| gatsby-plugin-clerk | 5.0.0-beta.45 |
| @clerk/nextjs | 5.0.10-snapshot.v58e76d6 |
| @clerk/clerk-react | 5.0.6-snapshot.v58e76d6 |
| @clerk/remix | 4.0.8-snapshot.v58e76d6 |
| @clerk/clerk-sdk-node | 5.0.7-snapshot.v58e76d6 |
| @clerk/shared | 2.1.1-snapshot.v58e76d6 |
| @clerk/testing | 1.0.5-snapshot.v58e76d6 |

Tip: Use the snippet copy button below to quickly install the required packages.
`@clerk/backend`
```sh
npm i @clerk/backend@1.1.5-snapshot.v58e76d6 --save-exact
```
`@clerk/chrome-extension`
```sh
npm i @clerk/chrome-extension@1.0.10-snapshot.v58e76d6 --save-exact
```
`@clerk/clerk-js`
```sh
npm i @clerk/clerk-js@5.3.1-snapshot.v58e76d6 --save-exact
```
`@clerk/elements`
```sh
npm i @clerk/elements@0.4.2-snapshot.v58e76d6 --save-exact
```
`@clerk/clerk-expo`
```sh
npm i @clerk/clerk-expo@1.1.2-snapshot.v58e76d6 --save-exact
```
`@clerk/express`
```sh
npm i @clerk/express@0.0.8-snapshot.v58e76d6 --save-exact
```
`@clerk/fastify`
```sh
npm i @clerk/fastify@1.0.7-snapshot.v58e76d6 --save-exact
```
`gatsby-plugin-clerk`
```sh
npm i gatsby-plugin-clerk@5.0.0-beta.45 --save-exact
```
`@clerk/nextjs`
```sh
npm i @clerk/nextjs@5.0.10-snapshot.v58e76d6 --save-exact
```
`@clerk/clerk-react`
```sh
npm i @clerk/clerk-react@5.0.6-snapshot.v58e76d6 --save-exact
```
`@clerk/remix`
```sh
npm i @clerk/remix@4.0.8-snapshot.v58e76d6 --save-exact
```
`@clerk/clerk-sdk-node`
```sh
npm i @clerk/clerk-sdk-node@5.0.7-snapshot.v58e76d6 --save-exact
```
`@clerk/shared`
```sh
npm i @clerk/shared@2.1.1-snapshot.v58e76d6 --save-exact
```
`@clerk/testing`
```sh
npm i @clerk/testing@1.0.5-snapshot.v58e76d6 --save-exact
```

**Comment by octoper** [comment]
!snapshot

**Review by nikosdouvlis — APPROVED** [review]
Nice one @octoper 👏🏻 
I will test a little bit more and I'll merge right after

**Review by BRKalow — APPROVED** [review]
👏

**Pull request merged**
